mobile/src/navigations/root/root.tsx

```diff
@@ -16,7 +20,40 @@ const screenOptions: NativeStackNavigationOptions = {
   headerShown: false,
 };
 
-const Root: React.FC = () => {
+type RootProperties = {
+  routeName: string | undefined;
+};
+
+const Root: React.FC<RootProperties> = ({ routeName }) => {
+  const dispatch = useAppDispatch();
+  const { authenticatedUserDataStatus } = useAppSelector(({ auth }) => ({
+    authenticatedUserDataStatus: auth.authenticatedUserDataStatus,
+  }));
+
+  const isRoot = routeName === RootScreenName.SIGN_IN;
+
+  useEffect(() => {
+    if (isRoot) {
+      void dispatch(userActions.loadAll()).catch((error: Error) => {
+        throw new Error(`Error loading all users: ${error.message}`);
+      });
+    }
+  }, [isRoot, dispatch]);
+
+  useEffect(() => {
+    void dispatch(authActions.getAuthenticatedUser()).catch((error: Error) => {
+      throw new Error(`Error getting authenticated user: ${error.message}`);
+    });
```

Comment:
We will have a common error handler for this
```suggestion
    void dispatch(authActions.getAuthenticatedUser());
```